migrations/2022-04-04-183652_update_community_aggregates_on_soft_delete/up.sql

```diff
@@ -0,0 +1,64 @@
+drop trigger if exists community_aggregates_post_count on post;
+drop trigger if exists community_aggregates_comment_count on comment;
+
+create or replace function community_aggregates_post_count()
+    returns trigger language plpgsql
+as $$
+begin
+  IF (TG_OP = 'INSERT') THEN
+update community_aggregates
+set posts = posts + 1 where community_id = NEW.community_id;
+ELSIF (TG_OP = 'DELETE'  OR (TG_OP = 'UPDATE' AND OLD.deleted = 'f' AND NEW.deleted = 't')) THEN
+update community_aggregates
+set posts = posts - 1 where community_id = OLD.community_id;
+
+-- Update the counts if the post got deleted
+update community_aggregates ca
```

Comment:
Might be a good idea to extract this into a function, and just run it any time the DELETE or UPDATE op is run.

Ideally also the trigger should only be run when those specific columns change. Otherwise it could cause performance issues. I think you set that in the trigger definition, not the function.

Reply:
Yep this should pry still be extracted to a function, sinces its used in several places.